### PR TITLE
feat(UI): build Radio component

### LINKED ISSUE #42: Create Base Radio component

### PR DESCRIPTION
**Pull request by mapra99**
Closes #42 

I built a component for radio inputs. I also built another one to group radio inputs in a list, to control the logic that allows the user to pick only one option from a given list of choices

image

image

### CODE REVIEW

File: src/renderer/components/atoms/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -13,6 +13,10 @@ const Checkbox = ({
   const [value, setValue] = useState<boolean>(checked || false);
   const inputId = id || `checkbox-${getRandomHex(16)}`;
 
+  useEffect(() => {
+    setValue(checked || false);
+  }, [checked]);
+
```
Comment (mapra99) [inline src/renderer/components/atoms/checkbox/checkbox.tsx:19]: I found a bug on this component, on `Toggle`, `Select` and `SelectionList` too. I'm trying to implement this pattern where we have some props that, when used, allow us to control the component from an outside state. The issue is that the internal state is not getting updated automatically, so this `useEffect` is making sure of that

File: src/renderer/components/atoms/radio/radio.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const inputId = id || `checkbox-${getRandomHex(16)}`;
+
+  useEffect(() => {
+    setIsChecked(checked || false);
+  }, [checked]);
+
+  const handleChange = () => {
+    setIsChecked(true);
+    if (onChange) onChange(true);
+  };
```
Comment (ImMPrada) [inline src/renderer/components/atoms/radio/radio.tsx:24]: Does it work when the Radio is a member of a group of options?

Comment (ImMPrada) [reply src/renderer/components/atoms/radio/radio.tsx:24]: Hahaha nevermind, I saw a group below and tested it

**Comment by ImMPrada** [comment]
Nice one!!!

**Review by ImMPrada — APPROVED** [review]

File: src/renderer/components/atoms/checkbox/checkbox.tsx (hunk 1 above)
Comment (ImMPrada) [reply src/renderer/components/atoms/checkbox/checkbox.tsx:19]: nice!

**Pull request merged**